python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -0,0 +1,594 @@
+    @pipeBase.timeMethod
+    def matchObjectsToSources(self, refCat, sourceCat, wcs, refFluxField,
+                              match_tolerance=None):
+        """!Match sources to position reference stars
+
+        @param[in] refCat  catalog of reference objects that overlap the
+        exposure; reads fields for:
+            - coord
+            - the specified flux field
+        @param[in] sourceCat  catalog of sources found on an exposure; reads
+        fields for:
+            - centroid
+            - centroid flag
+            - edge flag
+            - saturated flag
+            - aperture flux, if found, else PSF flux
+        @param[in] wcs  estimated WCS
+        @param[in] refFluxField  field of refCat to use for flux
+        @param[in] match_tolerance is a MatchTolerance class object or None.
+            This this class is used to comunicate state between AstrometryTask
+            and MatcherTask. AstrometryTask will also set the MatchTolerance
+            class variable maxMatchDist based on the scatter AstrometryTask has
+            found after fitting for the wcs.
+        @return an lsst.pipe.base.Struct with fields:
+        - matches  a list of matches, each instance of
+            lsst.afw.table.ReferenceMatch
+        - usableSourcCat  a catalog of sources potentially usable for
+            matching.
+        - match_tolerance a MatchTolerance object containing the resulting
+            state variables from the match.
+        """
+        import lsstDebug
+        debug = lsstDebug.Info(__name__)
+
+        preNumObj = len(refCat)
+        refCat = self.filterStars(refCat)
+        numRefObj = len(refCat)
+
+        if self.log:
+            self.log.info(
+                "filterStars purged %d reference stars, leaving %d stars" %
+                (preNumObj - numRefObj, numRefObj))
+
+        # If we get an empty tolerance struct create the variables we need for
+        # this matcher.
+        if match_tolerance is None:
+            match_tolerance = MatchTolerance()
+
+        # usableSourceCat: sources that are good but may be saturated
+        numSources = len(sourceCat)
+        selectedSources = self.sourceSelector.selectSources(sourceCat)
+        goodSourceCat = selectedSources.sourceCat
+        numUsableSources = len(goodSourceCat)
+        self.log.info("Purged %d sources, leaving %d good sources" %
+                      (numSources - numUsableSources, numUsableSources))
+
+        if len(goodSourceCat) == 0:
+            raise pipeBase.TaskError("No sources are good")
+
+        # avoid accidentally using sourceCat; use goodSourceCat from now on
+        del sourceCat
+
+        minMatchedPairs = min(self.config.minMatchedPairs,
+                              int(self.config.minFracMatchedPairs *
+                                  min([len(refCat), len(goodSourceCat)])))
+
+        # match usable (possibly saturated) sources and then purge saturated
+        # sources from the match list
```

Comment:
The comment has been removed. As you say, I no longer use saturated objects in the matcher.